src/ensembl/src/content/app/custom-download/components/checkbox-grid/CheckboxGrid.tsx

```diff
@@ -0,0 +1,146 @@
+import React from 'react';
+import Checkbox from 'src/shared/checkbox/Checkbox';
+
+import styles from './CheckBoxGrid.scss';
+
+type Props = {
+  gridData: any;
+  columns: number;
+  hideUnchecked?: boolean;
+  hideTitles?: boolean;
+  checkboxOnChange: (status: boolean, subSection: string, id: string) => void;
+};
+
+export const filterCheckedAttributes = (attributes: any) => {
+  const filteredAttributes: any = {};
+
+  Object.keys(attributes).forEach((section) => {
+    Object.keys(attributes[section]).forEach((attributeId) => {
+      if (attributes[section][attributeId].checkedStatus === true) {
+        if (!filteredAttributes[section]) {
+          filteredAttributes[section] = {};
+        }
+        filteredAttributes[section][attributeId] =
+          attributes[section][attributeId];
+      }
+    });
+  });
+
+  return filteredAttributes;
+};
+
+export const getAttributesCount = (attributes: any) => {
+  let totalAttributes = 0;
+
+  if (!attributes || Object.keys(attributes).length === 0) {
+    return 0;
+  }
+  Object.keys(attributes).forEach((section) => {
+    totalAttributes += Object.keys(attributes[section]).length;
+  });
+
+  return totalAttributes;
+};
+
+const renderCheckBoxList = (
+  checkboxList: any,
+  props: Props,
+  subSection: string
+) => {
+  if (!Object.keys(checkboxList).length) {
+    return null;
+  }
+
+  const checkboxListIDs = Object.keys(checkboxList).sort();
```

Comment:
Not that it's important, but probably better not repeat the same operation (`Object.keys(checkboxList)`) twice. Get the keys once and save them as a variable?

Reply:
That's done.